### PR TITLE
Fix use of docker removed ParseRepositoryTag() function

### PR DESCRIPTION
**Pull request by stefwalter**
Docker has removed the ParseRepositoryTag() function in
leading to failures using the kubernetes Go client API.

Failure:

```
../k8s.io/kubernetes/pkg/util/parsers/parsers.go:30: undefined: parsers.ParseRepositoryTag
```

### CODE REVIEW

File: pkg/util/parsers/parsers_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+package parsers
+
+import (
+	"testing"
+)
+
+// Removed from docker code, added here
+
+func TestParseImageName(t *testing.T) {
+	if repo, tag := ParseImageName("root"); repo != "root" || tag != "latest" {
```
Comment (brendandburns) [inline pkg/util/parsers/parsers_test.go]: Please do this as a table test instead.

File: pkg/util/parsers/parsers_test.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+limitations under the License.
+*/
+
+package parsers
+
+import (
+	"testing"
+)
+
+// Removed from docker code, added here
```
Comment (brendandburns) [inline pkg/util/parsers/parsers_test.go]: Reference code URL?

File: pkg/util/parsers/parsers.go [hunk 3]
```diff
@@ -17,17 +17,38 @@ limitations under the License.
...
 import (
-	"github.com/docker/docker/pkg/parsers"
+	"strings"
 )
 
 const (
 	defaultImageTag = "latest"
 )
 
+// Removed from docker code, added here.
```
Comment (brendandburns) [inline pkg/util/parsers/parsers.go]: Reference to the Docker code.

File: pkg/util/parsers/parsers_test.go (hunk 1 above)
Comment (stefwalter) [inline pkg/util/parsers/parsers_test.go]: Done.

File: pkg/util/parsers/parsers_test.go (hunk 2 above)
Comment (stefwalter) [inline pkg/util/parsers/parsers_test.go]: Added link to commit where this code was removed from docker repo.

File: pkg/util/parsers/parsers.go (hunk 3 above)
Comment (stefwalter) [inline pkg/util/parsers/parsers.go]: Added link to commit where this code was removed from docker repo.

**Comment by ncdc** [comment]
@stefwalter what command are you running that leads to this compilation error?

**Comment by ncdc** [comment]
2 things:

1. The code in this PR was originally committed to github.com/docker/docker, which means we'd need a CLA from the original author, according to @erictune 
2. In the docker commit that removed the function in question, it was replaced with `reference.ParseNamed` from github.com/docker/distribution. I think that's probably a better long-term change, since it won't require copying in someone else's code and therefore no third-party CLA consent is required.

**Comment by stefwalter** [comment]
>  what command are you running that leads to this compilation error?

Using pkg/client from out of tree.

> In the docker commit that removed the function in question, it was replaced with reference.ParseNamed from github.com/docker/distribution. 

I spent an hour trying to do this (building my own godep, since upstream version is incompatible with kubernetes, following the documentation in development.md, updating github.com/Sirupsen/logrus) ... but no matter what I try I always end up at:

```
$ godep save ./...
godep: Package (context) not found
```

go dependencies are a pain. Anyone up for some dependency solving?

**Comment by stefwalter** [comment]
There's a WIP commit here with the code changes necessary to make this happen. The Godeps changes are what needs help.

**Comment by ncdc** [comment]
I can help you with the Godeps issue. I'll give it a shot with your branch and let you know how it goes / what I did (assuming it works).

**Comment by stefwalter** [comment]
Thanks @ncdc . Just updated my branch now (in case you pulled it).

**Comment by ncdc** [comment]
Need to run the hack script to update godep licenses

**Comment by stefwalter** [comment]
> Need to run the hack script to update godep licenses

Done.

**Comment by brendandburns** [comment]
Not passing tests...

**Comment by brendandburns** [comment]
LGTM.

**Pull request merged**